app/beacon.py

```diff
@@ -70,11 +78,26 @@ def __init__(self, url, slots_per_epoch):
     @proxy_connect_timeout_exception
     def get_block_by_beacon_slot(self, slot):
         response = session.get(urljoin(self.url, self.api_beacon_block.format(slot)), timeout=DEFAULT_TIMEOUT)
+
+        if response.status_code == 404:
+            raise BeaconBlockNotFoundError()
+
         try:
             return int(response.json()['data']['message']['body']['execution_payload']['block_number'])
         except KeyError as error:
             logging.error(f'Response [{response.status_code}] with text: {str(response.text)} was returned.')
-            raise KeyError from error
+            raise error
+
+    def get_slot_for_report(self, ref_slot: int, epochs_per_frame: int, slots_per_epoch: int):
+        for slot_num in range(ref_slot, ref_slot - epochs_per_frame * slots_per_epoch + 1, -1):
```

Comment:
```suggestion
        for slot_num in range(ref_slot, ref_slot - epochs_per_frame * slots_per_epoch - 1, -1):
```
as it's a decreasing range

Reply:
- 1 would be ours, previous report

Reply:
Removed +1 at all